images/custom-error-pages/rootfs/main.go

```diff
@@ -86,15 +103,82 @@ func main() {
 		defaultFormat = os.Getenv(DefaultFormatVar)
 	}
 
-	http.HandleFunc("/", errorHandler(errFilesPath, defaultFormat))
+	isExportMetrics := true
+	if os.Getenv(IsExportMetricsVar) != "" {
+		val, err := strconv.ParseBool(os.Getenv(IsExportMetricsVar))
+		if err == nil {
+			isExportMetrics = val
+		}
+	}
+
+	metricsPort := "8080"
```

Comment:
would folks want configure this port?

Reply:
Good point. I believe one of the use cases mentioned in https://github.com/kubernetes/ingress-nginx/issues/9152 is to not expose /metrics to the public, but still let it be scraped internally. For such use case, I'm not really sure they would want to configure this port, so long as it is a different port from where custom error pages are served. It probably just needs to be properly documented, so that they can configure targets for scrapping.

What do you think?

Reply:
Actually, setting something like `METRICS_PUBLIC` or `METRICS_PRIVATE` seems a little bit weird to me when compared to `METRICS_PORT` -- was that what you meant? Either would still require changes to the Helm chart to expose /metrics. Is it safe to assume we wish to support the previously mentioned use case?

Reply:
Please take another look @strongjz 

Thank you

Reply:
I believe this conversation is resolved, please re-open if it is still relevant